### ISSUE TITLE
Language

### ISSUE THREAD

**Issue by gregoriopellegrino** [ways2read/InDesignA11Y#1]
Knowledge base reference: http://kb.daisy.org/publishing/docs/html/title.html

# Issue
InDesign's export to EPUB 3 does not correctly insert the `lang` (and `xml:lang`) attribute for exported XHTML files, specifically:
- for cover and toc it does not insert any lang attribute
- for the contents, it inserts the lang attributes in the `body` tag, instead of the `html` tag

# How to fix it
For all HTML files export the `lang` attribute and the `xml:lang` attribute in the `html` tag.

**Comment by ways2read** [comment]
Colleagues in Adobe Engineering suggested keeping the “lang” attribute in the `<body>` tag where it is currently, and not removing it to maintain backward compatibility in case it has usage elsewhere, and just adding it to `<html>` as required for the bug fix. So it’ll be present in 2 places.
 
Does that work or will it make it difficult for the assistive tech to read with being present at two places?

**Comment by LauraB7** [comment]
User story: "As a content creator the language is included correctly."

**Comment by CircularKen** [comment]
The language tag has now been moved to the correct place in XHTML pages EPUB exports from InDesign 18.5.
The language tag has still does not appear for the toc in InDesign 18.5.

**Comment by gregoriopellegrino** [comment]
After a thorough check:
- the attribute `lang` and `xml:lang` is missing in the file `toc.xhtml`
- the `xml:lang` attribute is missing in the `content.opf` file.

You can detect these issues using the [Ace](https://daisy.github.io/ace/) tool.

**Comment by raman211** [comment]
Hi @ways2read, @LauraB7 , @gregoriopellegrino, @CircularKen

We have released this feature as part of the pre release builds a week back and the same are available here with 
https://forums.adobeprerelease.com/indesigndev/discussion/430/rcs-builds-for-indesign-incopy-and-ids-2024-19-0-1-205-along-with-the-sdk-are-now-available#latest
https://forums.adobeprerelease.com/indesign/discussion/2377/rcs-builds-for-indesign-and-incopy-2024-19-0-1-205-builds-are-now-available-to-download#latest

Hope you have played with the feature and we are waiting to hear your feedback on it.

Thanks,
Ramanjulu.

**Comment by gregoriopellegrino** [comment]
@raman211 thank you. We have tested it and everything seems okay to us. With this release the work on this issue is concluded for us. We will close the issue once the feature is released in a public version of InDesign.

**Comment by CircularKen** [comment]
When more than one language is present on a fixed layout page, I am seeing word level language tags applied.

In the sample InDesign file (attached) there are no overrides, no character styles and the language is being applied at the paragraph style. This is CORRECTLY being applied to the resulting EPUB at the paragraph level.

![image](https://github.com/ways2read/InDesignA11Y/assets/17338941/671c5bce-71c1-45fa-a49f-ab18c60a4813)

However, it is ALSO being unnecessarily and disruptively applied at every word level SPAN tags.

![image](https://github.com/ways2read/InDesignA11Y/assets/17338941/0758f631-8b43-46b5-855f-f1b7cb2dcc0a)

Declaring the language on each word can cause an EPUB reader to pause on every word. Here is an example video showing this unwanted effect (put sound on).

https://github.com/ways2read/InDesignA11Y/assets/17338941/0c0d9709-63b5-4871-afc9-69d896a3ead6

Sample files have been shared here
[Dual language.zip](https://github.com/ways2read/InDesignA11Y/files/14181329/Dual.language.zip)

FYI this is related to the [Language Tags in HTML bug](https://forums.adobeprerelease.com/indesign/discussion/comment/17580) in the InDesign Prerelease group initially listed by @LauraB7 and that persisted throughout InDesign 2023 but was resolved in InDesign 2024.

@raman211 Would you like this information posted there also?

**Comment by gregoriopellegrino** [comment]
With the release of InDesign 19.2 this issue regarding the missing lang attribute in the <html> element of XHTML files and OPF file for us is solved. Thank you @raman211, we can close this issue.

@CircularKen thank you for your report, we would ask you to open a separate issue tagging it as "FXL" so we can continue the discussion.
